src/DistributedTracingManager.ts

```diff
@@ -0,0 +1,251 @@
+                deviceItemList,
+                { placeHolder: "Select device...", ignoreFocusOut: true, canPickMany: true },
+            );
+
+            if (selectedDevices !== undefined && selectedDevices.length > 0) {
+                deviceIds = selectedDevices.map((deviceItem) => deviceItem.deviceId);
+            }
+        } else {
+            deviceIds = [node.deviceNode.deviceId];
+        }
+
+        if (deviceIds.length === 0) {
+            return;
+        }
+
+        this._outputChannel.show();
+        await this.updateDistributedTracingSettingForDevices(deviceIds, iotHubConnectionString, updateType);
+
+        if (node instanceof DistributedTracingLabelNode) {
+            vscode.commands.executeCommand("azure-iot-toolkit.refresh", node);
+        } else if (node instanceof DistributedTracingSettingNode) {
+            vscode.commands.executeCommand("azure-iot-toolkit.refresh", node.parent);
+        } else {
+            vscode.commands.executeCommand("azure-iot-toolkit.refresh");
+        }
+    }
+
+    public async updateDistributedTracingSettingForDevices(deviceIds: string[], iotHubConnectionString: string, updateType: DistributedSettingUpdateType) {
+        let registry = iothub.Registry.fromConnectionString(iotHubConnectionString);
+
+        let mode: boolean = undefined;
+        let samplingRate: number = undefined;
+        let twin;
+
+        if (deviceIds.length === 1) {
+            twin = await Utility.getTwin(registry, deviceIds[0]);
+
+            if (twin.properties.desired[Constants.DISTRIBUTED_TWIN_NAME]) {
+                mode = Utility.parseDesiredSamplingMode(twin);
+                samplingRate = Utility.parseDesiredSamplingRate(twin);
+            }
+
+            if (updateType === DistributedSettingUpdateType.OnlySamplingRate) {
+                mode = undefined;
+            }
+
+            if (updateType === DistributedSettingUpdateType.OnlyMode) {
+                samplingRate = undefined;
+            }
+        }
+
+        if (updateType !== DistributedSettingUpdateType.OnlySamplingRate) {
+            let selectedItem: SamplingModeItem = await vscode.window.showQuickPick(
+                this.getSamplingModePickupItems(),
+                { placeHolder: "Select whether to enable/disable the distributed tracing...", ignoreFocusOut: true },
+            );
+            if (!selectedItem) {
+                return;
+            }
+            mode = selectedItem.distributedTracingEnabled;
+        }
+
+        if (updateType !== DistributedSettingUpdateType.OnlyMode) {
+            if (mode !== false) {
+                samplingRate = await this.promptForSamplingRate(`Enter sampling rate, integer within [0, 100]`, samplingRate);
+
+                if (samplingRate === undefined) {
+                    return;
+                }
+            }
+        }
+
+        TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingStartEvent);
```

Comment:
Put it into the early line in updateDistributedTracingSetting()

Reply:
fixed